structuredheadings/plugin.js

```diff
@@ -31,12 +38,39 @@
         }
       };
 
+      var getCurrentBlock = function (element) {
+        return element.getAscendant({
+          div: 1,
+          p: 1,
+          pre: 1,
+          address: 1,
+          h1: 1,
+          h2: 1,
+          h3: 1,
+          h4: 1,
+          h5: 1,
+          h6: 1
+        }, true);
+      };
+
+      var getPreviousHeader = function (element) {
+        return element.getPrevious(function (node) {
+          if (node.type === CKEDITOR.NODE_ELEMENT &&
```

Comment:
very minor, and maybe it sucks for breakpointing:  we can actually just return this conditional